src/recommendpolicy/helperFunctions.go

```diff
@@ -92,8 +91,12 @@ func generatePolicy(name, namespace string, labels LabelMap) ([]types.KnoxSystem
 		}
 	}
 
-	return policies, nil
+	if cfg.GetCfgMergePolicy() {
+		GroupedbyLabelPolicies := GroupPoliciesByLabel(policies, ms, name, namespace, labels)
```

Comment:
Why do we need to group these policies by label.? Ideally this `policies` array will only contain policies for a single namespace, deployment and label. For example, policies for `mysql` deployment in `wordpres-mysql` namespace with label `app=mysql`. So why should we again group them by label?